pkg/kubectl/autoscale_test.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"reflect"
+	"testing"
+
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestHPAGenerate(t *testing.T) {
+	tests := []struct {
+		name       string
+		min        int32
+		cpuPercent int32
```

Comment:
> Fields min and cpuPercent seem never used.

I guess not.

For `min`, the following test cases use it:

`"valid case"`

`name: "'max' must be greater than or equal to 'min'"`

`name: "'min' must be an integer if specified"`

For `cpuPercent`, the following test cases use it:

`"valid case"`

`"cpu-percent must be an integer if specified"`

Am I misunderstanding your meaning?